pkg/manifests/manifests_test.go

```diff
@@ -1299,6 +1303,34 @@ func TestPrometheusK8sRemoteWriteURLs(t *testing.T) {
 	}
 }
 
+func TestPrometheusK8sRemoteWriteConfigWithProxy(t *testing.T) {
+	c, err := NewConfigFromString(`prometheusK8s:
+  remoteWrite:
+    - url: https://test.remotewrite.com/api/write
+      remoteTimeout: 30s
+`, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{httpProxy: "http://test.proxy.com"}, NewAssets(assetsPath), &APIServerConfig{}, &configv1.Console{})
+	p, err := f.PrometheusK8s(
+		&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+		&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+		nil,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if p.Spec.RemoteWrite[0].URL != "https://test.remotewrite.com/api/write" {
+		t.Errorf("want remote write URL https://test.remotewrite.com/api/write, got %v", p.Spec.RemoteWrite[0].URL)
+	}
+	if p.Spec.RemoteWrite[0].ProxyURL != "http://test.proxy.com" {
```

Comment:
nit: maybe we could add a check to `TestPrometheusK8sConfiguration` that no proxy is set by default (who knows).